### ISSUE TITLE
provide env vars for flag arguments

### ISSUE THREAD

**Issue by rayjlinden** [moov-io/paygate#158]
We kinda/sorta follow the 12 factor app model and have everything that can be configured in a service configured by environment variable.  It is generally easier and more standard than setting configs via launch flags.  (Not all apps handle flags the same way.)

As such it would be nice if flags like:  -http.addr :80 -log.format json

could also be set via environment variable.
HTTP_ADDR=:80
LOG_FORMAT=json
etc.

**Comment by adamdecaf** [comment]
I've noticed this too. It's inconsistent and we can fix. Overall I'd like to see paygate read from a [file based config](https://github.com/moov-io/paygate/issues/130) in the future.

**Comment by rayjlinden** [comment]
We find env vars to be preferable.  We do manage them as .env files which
is useful.  Again, the main reason is it remains consistent across many
different types of services.

A file based config is certainly sometimes needed.  If config gets
complicated or hierarchal.  Indeed, many infrastructure type services use
them (elasticsearch, mysql, etc.)

For these we prefer if what is in the file is stuff that is not environment
specific.  Such that the file is something we just package in the
container.  But for things like URL references to other services, or
credentials we try to pass them in via environment variables.  In fact,
when we can't we end up writing a start.sh that generates a config file
from environment variables and then start the service.

Otherwise, the trick of volume mounting a config gets rather difficult.  We
either have to push for non-standard infrastructure set up - or what we we
usually do is store the entire file in the secrets store or an environment
variable and again create a start.sh file that loads that data and
generates the file before launching.

So bottom line - I'm voting to keep it all environment variables as long as
possible!  :)

**Comment by adamdecaf** [comment]
I created https://github.com/moov-io/paygate/pull/162 to resolve this. Thoughts?
